PYME/LMVis/pipeline.py

```diff
@@ -759,63 +762,49 @@ def OpenFile(self, filename= '', ds = None, **kwargs):
                 # This won't effect local file loading even if loading is lazy (i.e. shouldn't cause a regression)
                 ds = self._ds_from_file(fn, **kwargs)
 
-            
-        #wrap the data source with a mapping so we can fiddle with things
-        #e.g. combining z position and focus 
-        mapped_ds = tabular.MappingFilter(ds)
-
-        
-        if 'PixelSize' in kwargs.keys():
-            mapped_ds.addVariable('pixelSize', kwargs['PixelSize'])
-            mapped_ds.setMapping('x', 'x*pixelSize')
-            mapped_ds.setMapping('y', 'y*pixelSize')
-
-        #extract information from any events
-        self.ev_mappings, self.eventCharts = _processEvents(mapped_ds, self.events, self.mdh)
+        # skip the MappingFilter wrapping, etc. in self.addDataSource and add this datasource as-is
+        ds.mdh, ds.events = self.mdh, self.events
+        self.dataSources['results_source'] = ds
 
+        add_pipeline_variables = Pipelineify(self.recipe, 
+            inputFitResults='results_source',
+            inputDriftResults='' if self.driftInputMapping is None else self.driftInputMapping,
+            inputEvents='' if self.events is None else self.events,
```

Comment:
This doesn't work - `inputEvents` takes a string which is a key to the recipe namespace. We would need to inject the events into the recipe namespace (which is a bit gross).

Reply:
With all the events loading we do in file opening, do we actually need this here? If we need to maintain the option to pop in events from another file or something am I going really recipe crazy to suggest we write a module to load events and add them to data source? I guess events in the recipe namespace at all seems to be a bit of an api fudge

Reply:
Events in the recipe namespace is definitely a fudge (which is one of the reasons why I have some reservations about pipelinify in VisGUI), but was the quickest way of getting things working in bakeshop etc ... It also maps pretty well to the automatic (non h5r specific) HDF loading we do in recipes where we load every table from the hdf file into the namespace.

We need to get the events from somewhere in order to generate the derived columns ... in the current VisGUI loading workflow we avoid any injection nastiness by loading them once, but that isn't a super good match to pipelinify.